ethereum-common/src/calls.rs

```diff
@@ -92,3 +96,70 @@ pub fn call_keys(call: &substreams_ethereum::pb::eth::v2::Call) -> Vec<String> {
 
     keys
 }
+
+pub mod tests {
+    use super::*;
+
+    use crate::pb::sf::substreams::ethereum::v1::{Call, Calls};
+    use prost::Message;
+
+    use anyhow::Error;
+    use substreams_ethereum::pb::eth::v2::Block;
+
+    use base64::decode;
+    use std::fs;
+
+    #[test]
+    fn test_all_calls() {
+        let block = parse_block().expect("Failed to parse block");
+
+        let result = _all_calls(block).expect("Failed to execute function");
+        assert_eq!(result.calls.len(), 670);
+    }
+
+    #[test]
+    fn test_filtered_calls() {
+        // Given
+        let block = parse_block().expect("Failed to parse block");
+
+        let calls_obj = Calls {
+            calls: block
+                .calls()
+                .into_iter()
+                .map(|c| {
+                    return Call {
+                        tx_hash: "0x".to_owned(),
+                        call: Some(c.call.clone()),
+                    };
+                })
+                .collect(),
+            clock: None,
+        };
+
+        // When
+        let result = _filtered_calls(
+            "call_from:0x5acc84a3e955bdd76467d3348077d003f00ffb97".to_owned(),
+            calls_obj,
+        )
+        .expect("Failed to execute function");
+
+        // Expect
+        result.calls.iter().for_each(|c| {
+            let caller = &c.call.as_ref().unwrap().caller;
+
+            assert_eq!(
+                Hex::encode(&caller),
+                "5acc84a3e955bdd76467d3348077d003f00ffb97"
+            );
+        });
+    }
+
+    fn parse_block() -> Result<Block, Error> {
+        let encoded = fs::read_to_string("./src/test_block_10500500")?;
```

Comment:
Use `file!` macro to obtain current file and compute the correct path after.

```

    fn testdata_file(test_filename: &str) -> PathBuf {
        Path::parent(Path::new(file!()))
            .expect("Failed to get current source file directory")
            .join(test_filename)
    }
```

We could probably think about re-using for all tests within this.

Also for the file itself, let's put it in it's own subfolder, like `src/testdata` and ensure the network is in the filename, so it should look in the end like: `src/testdata/ethereum_mainnet_10500500.binpb.base64`